### PR TITLE
MAINT: clean up some macros in scalarapi.c

### PR DESCRIPTION
**Pull request by eric-wieser**
Using a macro in an if without parens is just weird.

Also, the void cases here match the pattern, so do not need special-casing.

### CODE REVIEW

**Comment by seberg** [comment]
Agreed about the brackets, could probably remove it from the `_CHK` macro. Unlike the simple cases, it returns a pointer instead of the value though (well, I suppose it cannot return a value, so maybe that makes sense).

**Comment by eric-wieser** [comment]
Sloppy of me, thanks for catching the pointer thing. Updated.

File: numpy/core/src/multiarray/scalarapi.c [hunk 1]
```diff
@@ -33,6 +33,8 @@ _descr_from_subtype(PyObject *type)
 NPY_NO_EXPORT void *
 scalar_value(PyObject *scalar, PyArray_Descr *descr)
 {
+#define _OBVAL(Name) (((Py##Name##ScalarObject *)scalar)->obval)
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/scalarapi.c]:
```suggestion
#define _OBVAL(Name) PyArrayScalar_VAL(scalar, Name)
```

**Comment by seberg** [comment]
LGTM, just waiting for tests, thanks.

**Comment by eric-wieser** [comment]
Glad I got there in the end - a little bit more work than I was intending to put into a throwaway patch that distracted me from #15385

**Pull request merged**